### PR TITLE
Add color noise transform

### PR DESCRIPTION
**Pull request by mmxgn**
Add `AddColorNoise`  which lets us add noise with _color_ (such as pink, violet, etc.) Additionally, we can A-weight the noise (and create e.g. grey noise).

### CODE REVIEW

**Comment by iver56** [comment]
Nice :) Can you also add it in demo.py?

**Comment by mmxgn** [comment]
> Nice :) Can you also add it in demo.py?

Done, also fixed support for multichannel (added test too)

**Comment by mmxgn** [comment]
Before merging it I want to decide on an issue:

In torch audiomentations the noise is computed once and is copied on all channels. 

I would prefer if it were independently. Tell me if that's ok. Additionally, should the noise power be distributed across the channels (e.g. in stereo), or not?

**Review by iver56 — APPROVED** [review]
LGTM and SGTM 👍 

Just two small requests:

* I would have renamed min_snr_in_db to min_snr_db and max_snr_in_db to max_snr_db. The plan is to do it like that in other transforms too: https://github.com/iver56/audiomentations/issues/177
* In the demo script, set `max_snr_in_db` to something higher than the default (40), so it becomes easier to evaluate/compare the different colors there. Or maybe consider using a higher default in AddColorNoise? I see that the default for max_snr_in_db in AddGaussianSNR is 40, while it is 30 in AddBackgroundNoise. We can of course adjust the default in AddGaussianSNR if we want to be more consistent across transforms. What do you think?

File: audiomentations/augmentations/add_color_noise.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        Color          f_decay
+        ----------------------
+        white              0.0
+        pink               1.0
+        brown(ian)/red     2.0
+        blue/azure        -1.0
+        violet            -2.0
+
+        Additionally, p_apply_a_weighting gives the probability the noise to be weighted by a psychoacoustic
+        equal loudness curve, which results in grey-noise when f_decay = 0.0.
```
[the remark this answers is not in the document]
Comment (mmxgn) [reply audiomentations/augmentations/add_color_noise.py:121]: I think I will keep the PR open a little bit more to add some examples in the docs, together with that information.

File: audiomentations/augmentations/add_color_noise.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        min_f_decay: float = -2.0,
+        max_f_decay: float = 2.0,
+        p_apply_a_weighting: float = 0.0,
+        p: float = 0.5,
+        n_fft: int = 128,
+    ):
+        """
+        :param min_snr_in_db: Minimum signal-to-noise ratio in dB. A lower number means more noise.
+        :param max_snr_in_db: Maximum signal-to-noise ratio in dB. A greater number means less noise.
+        :param min_f_decay: Minimum frequency decay in dB.
```
[the remark this answers is not in the document]
Comment (mmxgn) [reply audiomentations/augmentations/add_color_noise.py:98]: I used it to be the same as `torch-audiomentations`:

https://github.com/asteroid-team/torch-audiomentations/blob/2029547c7186794c20f7da3bdca815163f722f07/torch_audiomentations/augmentations/colored_noise.py#L13

 But I would prefer it as dB/octave too. Tell me what you'd prefer.

[the remark this answers is not in the document]
Comment (iver56) [reply audiomentations/augmentations/add_color_noise.py:98]: Hmm, dB/octave seems easier to understand, so I think I would prefer that

**Comment by iver56** [comment]
Any thoughts on the timeline for this?

**Comment by mmxgn** [comment]
As soon as I have time (tm)

**Comment by mmxgn** [comment]
Should I merge this? I can add the documentation at a later stage.

**Comment by iver56** [comment]
Here's the TODO list of things to do before I think it's ready for merge

- [ ] fix merge conflicts
- [ ] rename min_snr_in_db to min_snr_db (I have already changed to this naming convention in the rest of audiomentations)
- [ ] f decay in dB per octave instead of dB (best to do it now, or else there will be a breaking API change later

As you say, proper documentation can be postponed until a later PR

**Comment by mmxgn** [comment]
@iver56  can you check it does what you want now?

**Pull request merged**